### ISSUE TITLE
Pseudocode review - Nick Wagner

### ISSUE THREAD

**Issue by Nwagner22** [2019-bgmp/demultiplexing-paulaberry#2]
**Define the problem**: You never defined the problem at the top. This helps to show that you know exactly what needs to be accomplished in the code below.

**Output**: Same for output. A good example for output would have been to define the 52 
 total output files and and say the reason for each. Again to show that you know the requirements of the code and that you know what will be outputted. I see below now where you create all 52 output files, but still would be nice to have at the top where you are defining your problem.

**Functions and Unit tests**: Great job here with the stubbed out functions and the examples within each to show how they actually work and what they output. 

**Pseudocode algorithm**: Nice job with the pseudocode! Very easy to see the order in which you do your steps for the algorithm, and I really like the detail you add especially when defining the keys and values for your dictionaries. Another good thing to do would have been to add a text version of your pseudocode above so you can make sure you cover all parts of your algorithm that you define.
```
# After we have access to the current record in each file, we then want to test for each of the four possibilities before we can determine if it is dual matched
        # 1. if either of the indexes have Ns, put the whole record in the respective undetermined files
        # 2. if the phred score of any of the index bases is below the coverage cutoff, put the whole record in the respective undetermined files
        # 3. if the forward index or the reverse-compliment of the reverse index do not match the 24 given indexes, put the whole record in the respective undetermined files
        # 4. if both are valid indexes, but are not the reverse compliment of eachother, then the whole record is put in the respective index hopped files
```

Overall great job though!
